src/client/MealsWithID.js

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect } from 'react'
+import { useParams } from "react-router-dom";
+import ShowReservations from './ShowReservations.js'
+
+import images from './images.js'
+import './MealsWithID.css'
+
+const API = "/api/meals"
+
+
+const MealsWithId = ({ meals, setMeals }) => {
+
+    useEffect(() => {
+        async function DataFetched() {
+            const fetchdata = await fetch(API)
+            const responseFetch = await fetchdata.json()
+            setMeals(responseFetch)
+        }
+
+        DataFetched()
+    }, [])
+
+    const params = useParams()
+    console.log(meals)
+    const meal = meals.find((meal) => meal.id == Number(params.id))
+    if (meal === undefined) {
```

Comment:
Think about your error handling here also. Instead of "return null" you should find a way to display the error to the user. Think about how you could create some generic/reusable logic here